tests/countries/test_sao_tome_and_principe.py

```diff
@@ -0,0 +1,126 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.sao_tome_and_principe import SaoTomeAndPrincipe
+from tests.common import CommonCountryTests
+
+
+class TestSaoTomeAndPrincipe(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(SaoTomeAndPrincipe, years=range(2014, 2050))
+        cls.observed_holidays = SaoTomeAndPrincipe(years=range(2014, 2050), observed=True)
+
+    def test_new_years_day(self):
+        for year in range(2014, 2050):
+            self.assertHolidayName("Ano Novo", f"{year}-01-01")
+
+    def test_king_amador_day(self):
+        for year in range(2014, 2050):
+            self.assertHolidayName("Dia do Rei Amador", f"{year}-01-04")
```

Comment:
For standalone holidays that are observed annually:
```suggestion
    def test_day_of_king_amador(self):
        self.assertHolidayName("Dia do Rei Amador", (f"{year}-01-04" for year in range(2014, 2050)))
```
It’s best to keep the test case names as close as possible to the en_US holiday names to avoid confusion and improve clarity.